single_tech_samples/synapseanalytics/sample3_integration_testing/tests/integrationtests/dataconnectors/adls.py

```diff
@@ -0,0 +1,231 @@
+    container: str,
+    file_path: str,
+    file_name: str,
+    base_path: str = None,
+    file_contents: bytes = None,
+):
+    """Upload file to ADLS either by reading a local file or directly
+    uploading the file contents
+
+    Args:
+        adls_connection_client (DataLakeServiceClient): ADLS Connection Object
+        container (str): Container Name where file needs to be uploaded
+        file_path (str): File Path of local file to upload
+        file_name (str): File Name of local file to upload
+        base_path (str): Base Folder Path in ADLS where file will be uploaded
+        file_contents (bytes): Contents of the file
+    Returns:
+        request_id (str): Id of the ADLS upload request
+    """
+
+    print(
+        "STARTING TO UPLOAD FILE TO ADLS"
+        f"account: {adls_connection_client.url} "
+        f"and container: {container}..."
+    )
+
+    try:
+        file_system_client = adls_connection_client.get_file_system_client(
+            file_system=container
+        )
+
+        if not base_path:
+            file_client = file_system_client.create_file(f"{file_name}")
+        else:
+            file_client = file_system_client.get_directory_client(
+                base_path
+            ).create_file(f"{file_name}")
+
+        if file_contents is None:
+            file_name, file_contents = local_file.read_local_file(file_path, file_name)
+
+        file_client.append_data(data=file_contents, offset=0, length=len(file_contents))
+        response = file_client.flush_data(
+            len(file_contents), close=True
+        )  # close=True must be set to work with the Synapse Trigger's Topic Advanced Filter
+        request_id = response.get("request_id")
+        print(f"SUCCESSFULLY UPLOADED {file_name} with id {request_id} to ADLS")
+        return request_id
+    except Exception as e:
+        print(e)
+        raise
+
+
+def download_from_ADLS(
+    adls_connection_client: DataLakeServiceClient,
+    container: str,
+    file_name: str,
+    base_path: str = None,
+):
+    """Download file from ADLS
+
+    Args:
+        adls_connection_client (DataLakeServiceClient): ADLS Connection Object
+        container (str): Container Name where file needs to be uploaded
+        base_path (str): Base Folder Path in ADLS where file will be uploaded
+        file_name (str): File Name to Upload
+
+    Returns:
+        downloaded_bytes (bytes): Contents of the file
+    """
+
+    print(
+        f"STARTING TO DOWNLOAD FILE {file_name}"
+        f"FROM ADLS account: {adls_connection_client.url} "
+        f"and container: {container}..."
+    )
+
+    try:
+        file_system_client = adls_connection_client.get_file_system_client(
+            file_system=container
+        )
+        if not base_path:
+            directory_client = file_system_client.get_directory_client("/")
+        else:
+            directory_client = file_system_client.get_directory_client(base_path)
```

Comment:
NIT - 

this can be reduced to avoid duplicating the 'get_directory_client' method usage to 1 by initializing the base)_path to "/" if it not set.

```
if not base_path:
   base_path = "/"
directory_client = file_system_client.get_directory_client(base_path)
```

Reply:
Thanks for catching this! I updated the default value for the parameter.